resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -793,6 +796,8 @@ private[spark] class ApplicationMaster(
     override def receive: PartialFunction[Any, Unit] = {
       case UpdateDelegationTokens(tokens) =>
         SparkHadoopUtil.get.addDelegationTokens(tokens, sparkConf)
+
+      case Shutdown => shutdown = true
```

Comment:
@tgravescs  move Shutdown to `receive` since it's a OneWayMessage.

And have tested this pr  in  our prod by use gdb to close socket between driver and am.
The behavior is as expected.